### ISSUE TITLE
Material v7 breaks mkdocs-localsearch

### ISSUE THREAD

**Issue by wilhelmer** [squidfunk/mkdocs-material#2372]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

See https://github.com/wilhelmer/mkdocs-localsearch/issues/12.

Does not occur with any of the v7 beta versions, only in the release versions.

### Steps to reproduce the bug

1. Create a blank MkDocs project.
2. Install and set up mkdocs-localsearch as described in the [docs](https://github.com/wilhelmer/mkdocs-localsearch/blob/master/README.md).
3. Build.

[Download sample project](https://github.com/squidfunk/mkdocs-material/files/6063775/localsearch-test.zip)

### Project configuration

``` yaml
site_name: My Docs
theme:
  name: material
  custom_dir: theme
use_directory_urls: false
plugins:
  - search
  - localsearch
```

### System information

* OS: macOS
* Browser: Firefox and Chrome, latest

**Comment by squidfunk** [comment]
Well. I'm stupid:

https://github.com/squidfunk/mkdocs-material/blob/9ed181b697537dc71407ab4529e2791cdfbf8b23/src/assets/javascripts/components/search/_/index.ts#L87-L88

I added this because I wanted the search to degrade gracefully (= just hide) when served on `file://`, forgetting about the `localsearch` plugin 🤦‍♂️ Fixed in eb35ab94.

**Comment by squidfunk** [comment]
BTW, if you find some time, you could check if the Promise delay is still necessary. It shouldn't be. If it is and you could provide another reproducible example, we could finally fix it.

**Comment by wilhelmer** [comment]
Thanks a lot, once again. 

> I added this because I wanted the search to degrade gracefully (= just hide) when served on `file://`, forgetting about the `localsearch` plugin 🤦‍♂️

I think it's a great idea to hide search when accessed locally, you should stick to that. Maybe keep the switch but introduce some flag that `localsearch` can set? Something like `app.search$.isLocal`?

> BTW, if you find some time, you could check if the Promise delay is still necessary. It shouldn't be. If it is and you could provide another reproducible example, we could finally fix it.

I'll look into that.

**Comment by squidfunk** [comment]
> I think it's a great idea to hide search when accessed locally, you should stick to that. Maybe keep the switch but introduce some flag that localsearch can set? Something like app.search$.isLocal?

That's one idea, but on the other hand, building your docs and distributing them as HTML you should either deliberately disable search or opt-in to using localsearch, so this is what's keeping me from it. The behavior would be more consistent, and involve less "magic" 😅 This is also documented in the "offline search" section of the docs.

**Comment by wilhelmer** [comment]
Tried with a 35 MB search index without the Promise delay, worked perfectly. I've removed the setting and released [mkdocs-localsearch 0.9.0](https://github.com/wilhelmer/mkdocs-localsearch/releases/tag/0.9.0).

**Comment by squidfunk** [comment]
Perfect! Thanks for investigating!

**Comment by wilhelmer** [comment]
On a side note, if search is disabled completely (by omitting the `search` plugin in mkdocs.yml), Material v7 throws exceptions, which v6 didn't do:

```console
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at file:///C:/.../site/search/search_index.json. (Reason: CORS request not http).

Uncaught (in promise) TypeError: NetworkError when attempting to fetch resource.
    Ht file:///C:/.../site/assets/javascripts/bundle.024964f1.min.js:3
    ye file:///C:/.../site/assets/javascripts/bundle.024964f1.min.js:3
    <anonymous> file:///C:/.../site/assets/javascripts/bundle.024964f1.min.js:81
    <anonymous> file:///C:/.../site/assets/javascripts/bundle.024964f1.min.js:81
bundle.02
```

**Comment by squidfunk** [comment]
Definitely an error, thanks for reporting! I'll address this before issuing a new release.

**Comment by squidfunk** [comment]
Fixed in 5a510ced.

**Comment by squidfunk** [comment]
Released with 7.0.4
